src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,88 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as currentTodoActions } from '../../features/todoReducer';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+  const [userFromServer, setUserFromServer] = useState<User | null>(null);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
 
-      <Loader />
+  const closeModal = () => dispatch(currentTodoActions.removeTodo());
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  useEffect(() => {
+    if (currentTodo?.userId) {
+      getUser(currentTodo.userId)
+        .then(setUserFromServer)
+        .catch(() => {
+          setUserFromServer(null);
+        });
+    }
+  }, [currentTodo]);
+
+  if (!currentTodo) {
+    return null;
+  }
+
+  const renderUserInfo = () => {
+    if (!userFromServer) {
+      return <p>Loading user...</p>;
```

Comment:
Consider using the `Loader` component here instead of a plain text message to maintain consistency in the loading experience across the application.